File: Challenge 2/challenge2.py
class Employee:
    def __init__(self, name, employee_type, hours_worked, hourly_rate, bonus=0):
        self.name = name
        self.type = employee_type
        self.hours_worked = hours_worked
        self.hourly_rate = hourly_rate
        self.bonus = bonus


class PayrollCalculator:
    def calculate_salary(self, employee):
        salary = 0

        if employee.type == "FullTime":
            salary = employee.hours_worked * employee.hourly_rate + employee.bonus
        elif employee.type == "PartTime":
            salary = employee.hours_worked * employee.hourly_rate
        elif employee.type == "Intern":
            salary = employee.hours_worked * (employee.hourly_rate / 2)

        return salary


if __name__ == "__main__":
    employees = [
        Employee(name="John", employee_type="FullTime", hours_worked=40, hourly_rate=25, bonus=500),
        Employee(name="Jane", employee_type="PartTime", hours_worked=20, hourly_rate=20, bonus=0),
        Employee(name="Alice", employee_type="Intern", hours_worked=30, hourly_rate=15, bonus=0)
    ]

    payroll_calculator = PayrollCalculator()

    for employee in employees:
        print(f"{employee.name}'s salary: {payroll_calculator.calculate_salary(employee)}")

File: Challenge 2/challenge2.cs
public class Employee {
    public string Name;
    public string Type;
    public double HoursWorked;
    public double HourlyRate;
    public double Bonus;
}

public class PayrollCalculator {
    public double CalculateSalary(Employee employee) {
        double salary = 0;

        if (employee.Type == "FullTime") {
            salary = employee.HoursWorked * employee.HourlyRate + employee.Bonus;
        }
        else if (employee.Type == "PartTime") {
            salary = employee.HoursWorked * employee.HourlyRate;
        }
        else if (employee.Type == "Intern") {
            salary = employee.HoursWorked * (employee.HourlyRate / 2);
        }

        return salary;
    }
}

public class Program {
    public static void Main() {
        var employees = new List<Employee> {
            new Employee { Name = "John", Type = "FullTime", HoursWorked = 40, HourlyRate = 25, Bonus = 500 },
            new Employee { Name = "Jane", Type = "PartTime", HoursWorked = 20, HourlyRate = 20, Bonus = 0 },
            new Employee { Name = "Alice", Type = "Intern", HoursWorked = 30, HourlyRate = 15, Bonus = 0 }
        };

        var payrollCalculator = new PayrollCalculator();

        foreach (var employee in employees) {
            Console.WriteLine($"{employee.Name}'s salary: {payrollCalculator.CalculateSalary(employee)}");
        }
    }
}
